Sources/Overture/Update.swift

```diff
@@ -39,6 +41,7 @@ public func update<A>(_ a: A, _ fs: ((inout A) -> Void)...) -> A {
 ///   - fs: In-out functions.
 /// - Returns: The result of `f` applied to `a`.
 /// - Note: This function is commonly seen in operator form as "pipe-forward", `|>`.
+@discardableResult
 public func update<A>(_ a: A, _ fs: ((inout A) throws -> Void)...) throws -> A {
```

Comment:
These are intended for value types, which get copied upon entry, and the copy is returned, so we definitely want to warn folks if they don't use the return value!

Please note that we have these alternative APIs for objects, which _are_ annotated: https://github.com/pointfreeco/swift-overture/blob/edfb7a1e84e0c93578be54e9fd3b37d7cbbf2962/Sources/Overture/Update.swift#L48-L72